src/host/consoleInformation.cpp

```diff
@@ -236,7 +238,25 @@ COLORREF CONSOLE_INFORMATION::GetDefaultForeground() const noexcept
 // - the default background color of the console.
 COLORREF CONSOLE_INFORMATION::GetDefaultBackground() const noexcept
 {
-    return Settings::CalculateDefaultBackground();
+    const auto bg = GetDefaultBackgroundColor();
+    return bg != INVALID_COLOR ? bg : GetColorTableEntry((LOBYTE(GetFillAttribute()) & BG_ATTRS) >> 4);
+}
+
+// Method Description:
+// - Get the colors of a particular text attribute, using our color table,
+//      and our configured default attributes.
+// Arguments:
+// - attr: the TextAttribute to retrieve the foreground color of.
+// Return Value:
+// - The color values of the attribute's foreground and background.
+std::pair<COLORREF, COLORREF> CONSOLE_INFORMATION::LookupAttributeColors(const TextAttribute& attr) const noexcept
+{
+    _blinkingState.RecordBlinkingUsage(attr);

```

Comment:
clever clearing it during a render cycle on-blink and using the function that resolves the color to tell whether we should keep blinking.

It took me a little bit to understand what was going on (and why I was worried that we would keep doing a full redraw every other cursor blink for all eternity), but... I like it.

It might want to be documented in a comment (sorry if it already has: I'm reading bottom-up).

Reply:
Yeah, I've tried to explain this in the doc comment for `RecordBlinkingUsage` and then there's more detail in the internal comments of `ToggleBlinkingRendition` as well.